### PR TITLE
tests: testify some server tests

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: ref #4813.

### What is changed and how does it work?

```commit-message
Testify some server tests.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/server/join/join_test.go [hunk 1]
```diff
@@ -21,157 +21,144 @@ import (
...
 
 	client := cluster.GetServer("pd1").GetEtcdClient()
 	_, err = client.MemberRemove(context.TODO(), pd3.GetServerID())
-	c.Assert(err, IsNil)
+	re.NoError(err)
 
 	// The server should not successfully start.
 	res := cluster.RunServer(pd3)
-	c.Assert(<-res, NotNil)
+	re.NotNil(<-res)
```
Comment (rleungx) [inline tests/server/join/join_test.go:107]:
```suggestion
	re.Error(<-res)
```

File: tests/server/join/join_test.go [hunk 2]
```diff
@@ -21,157 +21,144 @@ import (
...
+	re.NoError(err)
 
 	err = pd3.Stop()
-	c.Assert(err, IsNil)
+	re.NoError(err)
 
 	// The server should not successfully start.
 	res := cluster.RunServer(pd3)
-	c.Assert(<-res, NotNil)
+	re.NotNil(<-res)
```
Comment (rleungx) [inline tests/server/join/join_test.go:139]: ditto

File: tests/server/join/join_test.go [hunk 3]
```diff
@@ -21,157 +21,144 @@ import (
...
-	c.Assert(err, IsNil)
-	err = pd2.Destroy()
-	c.Assert(err, IsNil)
-	c.Assert(join.PrepareJoinCluster(pd2.GetConfig()), NotNil)
+	pd2, err := cluster.Join(ctx)
+	re.NoError(err)
+	re.NoError(pd2.Run())
+	re.NoError(pd2.Stop())
+	re.NoError(pd2.Destroy())
+	re.NotNil(join.PrepareJoinCluster(pd2.GetConfig()))
```
Comment (rleungx) [inline tests/server/join/join_test.go:163]:
```suggestion
	re.Error(join.PrepareJoinCluster(pd2.GetConfig()))
```

File: tests/server/region_syncer/region_syncer_test.go [hunk 4]
```diff
@@ -135,122 +117,129 @@ func (s *regionSyncerTestSuite) TestRegionSyncer(c *C) {
...
-	testutil.WaitUntil(c, func() bool {
+	re.Len(cacheRegions, regionLen)
+	testutil.Eventually(re, func() bool {
+		assert := assert.New(t)
 		for _, region := range cacheRegions {
 			r := followerServer.GetServer().GetBasicCluster().GetRegion(region.GetID())
-			if !(c.Check(r.GetMeta(), DeepEquals, region.GetMeta()) &&
-				c.Check(r.GetStat(), DeepEquals, region.GetStat()) &&
-				c.Check(r.GetLeader(), DeepEquals, region.GetLeader())) {
+			if !(assert.Equal(region.GetMeta(), r.GetMeta()) &&
```
Comment (rleungx) [inline tests/server/region_syncer/region_syncer_test.go:127]: Can we use `re` directly?

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**